test/e2e/specs/editor/various/writing-flow.spec.js

```diff
@@ -0,0 +1,1004 @@
+		await page.keyboard.type( '3' );
+		await pageUtils.pressKeyTimes( 'ArrowUp', 2 );
+		await pageUtils.pressKeyTimes( 'Delete', 2 );
+		await expect.poll( editor.getEditedPostContent )
+			.toBe( `<!-- wp:paragraph -->
+<p>1</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>3</p>
+<!-- /wp:paragraph -->` );
+
+		await page.keyboard.press( 'Delete' );
+
+		await expect.poll( editor.getEditedPostContent )
+			.toBe( `<!-- wp:paragraph -->
+<p>13</p>
+<!-- /wp:paragraph -->` );
+	} );
+
+	test( 'should preserve horizontal position when navigating vertically between blocks', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'abc' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '23' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowDown' );
+		await page.keyboard.type( '1' );
+
+		await expect.poll( editor.getEditedPostContent )
+			.toBe( `<!-- wp:paragraph -->
+<p>abc</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>123</p>
+<!-- /wp:paragraph -->` );
+	} );
+
+	test( 'should remember initial vertical position', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '1' );
+		await page.keyboard.press( 'Enter' );
+		await pageUtils.pressKeyWithModifier( 'shift', 'Enter' );
+		await page.keyboard.type( '2' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.type( 'x' ); // Should be right after "1".
+
+		await expect.poll( editor.getEditedPostContent )
+			.toBe( `<!-- wp:paragraph -->
+<p>1x</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p><br>2</p>
+<!-- /wp:paragraph -->` );
+	} );
+
+	test( 'should navigate contenteditable with side padding', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.press( 'Enter' );
+		await page.evaluate( () => {
+			document.activeElement.style.paddingLeft = '100px';
+		} );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.type( '1' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should extend selection into paragraph for list with longer last item', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'a' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '* b' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'cd' );
+
+		// Selects part of the first list item, although invisible.
+		await pageUtils.pressKeyWithModifier( 'shift', 'ArrowUp' );
+		await page.evaluate( () => new Promise( window.requestIdleCallback ) );
```

Comment:
This is a bit weird and looks like an implementation detail. Any idea how to refactor this? I guess we needed this because we want to make sure the cursor is at the correct position. Maybe we should just do that explicitly?

Will be nice if we can refactor this but I don't think this is blocking the merge.

Reply:
This test was passing for me locally even when I removed this line.

I will check possible ways we can refactor this.

Reply:
Based on my tests, it looks fine to remove this line, but local and CI test runs aren't always the same.

I also remembered that Riad had to use similar escape hatches in #32765 to get tests passing with React 18. So maybe we should audit all these hacks together and find a refactoring solution we can later document in migration practices.

Reply:
Should we merge this PR?